File: wfm-develop mobile 07.05.2026/wfm/mobile/ios/WFMUI/Sources/WFMUI/Components/WFMToast.swift
import SwiftUI

// MARK: - Toast Types

/// Тип тоста (определяет содержимое)
public enum WFMToastType {
    /// Только текст
    case text
    /// Текст + кнопка-ссылка
    case textWithButton(buttonTitle: String, action: () -> Void)
}

/// Состояние тоста (определяет цвет фона)
public enum WFMToastState {
    /// Стандартный (тёмный фон)
    case `default`
    /// Ошибка (красный фон)
    case error
}

// MARK: - Toast Data

/// Данные для отображения тоста
public struct WFMToastData: Identifiable {
    public let id = UUID()
    public let message: String
    public let type: WFMToastType
    public let state: WFMToastState

    public init(message: String, type: WFMToastType = .text, state: WFMToastState = .default) {
        self.message = message
        self.type = type
        self.state = state
    }
}

// MARK: - WFMToast View

/// Компонент Toast (всплывающее уведомление)
///
/// Использование:
/// ```swift
/// WFMToast(data: WFMToastData(message: "Смена открыта"))
/// WFMToast(data: WFMToastData(message: "Ошибка", state: .error))
/// WFMToast(data: WFMToastData(
///     message: "Задача назначена",
///     type: .textWithButton(buttonTitle: "Перейти", action: { })
/// ))
/// ```
public struct WFMToast: View {
    @Environment(\.wfmColors) private var colors

    public let data: WFMToastData

    public init(data: WFMToastData) {
        self.data = data
    }

    private var backgroundColor: Color {
        switch data.state {
        case .default: return colors.toastBg
        case .error:   return colors.toastBgError
        }
    }

    public var body: some View {
        HStack(spacing: WFMSpacing.s) {
            Text(data.message)
                .wfmBody15Regular()
                .foregroundColor(colors.toastText)
                .multilineTextAlignment(.leading)
                .frame(maxWidth: .infinity, alignment: .leading)

            if case .textWithButton(let title, let action) = data.type {
                Button(action: action) {
                    Text(title)
                        .font(WFMTypography.headline12Medium)
                        .foregroundColor(colors.toastText)
                }
                .buttonStyle(.plain)
            }
        }
        .padding(.horizontal, WFMSpacing.l)
        .padding(.vertical, WFMSpacing.m)
        .background(backgroundColor)
        .cornerRadius(WFMRadius.m)
    }
}

File: wfm-develop mobile 07.05.2026/wfm/mobile/android/ui/src/main/kotlin/com/beyondviolet/wfm/ui/components/WfmToast.kt
package com.beyondviolet.wfm.ui.components

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.slideInVertically
import androidx.compose.animation.slideOutVertically
import androidx.compose.animation.fadeIn
import androidx.compose.animation.fadeOut
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.beyondviolet.wfm.ui.theme.LocalWfmColors
import com.beyondviolet.wfm.ui.theme.WfmTypography
import com.beyondviolet.wfm.ui.theme.*

// MARK: - Типы

/** Тип тоста (определяет содержимое) */
sealed class WfmToastType {
    /** Только текст */
    object Text : WfmToastType()

    /** Текст + кнопка-ссылка */
    data class TextWithButton(
        val buttonTitle: String,
        val action: () -> Unit
    ) : WfmToastType()
}

/** Состояние тоста (определяет цвет фона) */
enum class WfmToastState {
    /** Стандартный (тёмный фон) */
    DEFAULT,
    /** Ошибка (красный фон) */
    ERROR
}

/** Данные для отображения тоста */
data class WfmToastData(
    val message: String,
    val type: WfmToastType = WfmToastType.Text,
    val state: WfmToastState = WfmToastState.DEFAULT
)

// MARK: - Компонент

/**
 * Toast-уведомление дизайн-системы WFM.
 *
 * Использование через [WfmToastHost]:
 * ```kotlin
 * WfmToastHost(toastManager = toastManager)
 * ```
 *
 * Показ тоста из ViewModel:
 * ```kotlin
 * toastManager.show("Смена открыта")
 * toastManager.show("Ошибка", state = WfmToastState.ERROR)
 * toastManager.show("Задача назначена", type = WfmToastType.TextWithButton("Перейти") { navigate() })
 * ```
 */
@Composable
internal fun WfmToastContent(data: WfmToastData, modifier: Modifier = Modifier) {
    val colors = LocalWfmColors.current

    val bgColor = when (data.state) {
        WfmToastState.DEFAULT -> colors.toastBg
        WfmToastState.ERROR   -> colors.toastBgError
    }

    Surface(
        modifier = modifier.fillMaxWidth(),
        shape = RoundedCornerShape(10.dp),
        color = bgColor
    ) {
        Row(
            modifier = Modifier.padding(horizontal = 16.dp, vertical = 12.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = data.message,
                style = WfmTypography.Body15Regular,
                color = colors.toastText,
                modifier = Modifier.weight(1f)
            )

            if (data.type is WfmToastType.TextWithButton) {
                TextButton(onClick = data.type.action) {
                    Text(
                        text = data.type.buttonTitle,
                        style = WfmTypography.Headline12Medium,
                        color = colors.toastText
                    )
                }
            }
        }
    }
}

// MARK: - Host

/**
 * Контейнер для тостов. Размещается в корневом Scaffold приложения.
 *
 * Для показа тостов поверх BottomSheet используйте параметр `toastManager`
 * в компоненте WfmBottomSheet.
 *
 * ```kotlin
 * Box(modifier = Modifier.fillMaxSize()) {
 *     // Контент приложения
 *     WfmToastHost(toastManager = toastManager)
 * }
 * ```
 */
@Composable
fun WfmToastHost(
    toastManager: ToastManager,
    modifier: Modifier = Modifier
) {
    val isVisible by toastManager.isVisible.collectAsState()
    val current by toastManager.current.collectAsState()

    Box(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp)
            .padding(bottom = 8.dp),
        contentAlignment = Alignment.BottomCenter
    ) {
        AnimatedVisibility(
            visible = isVisible && current != null,
            enter = slideInVertically(initialOffsetY = { it }) + fadeIn(),
            exit = slideOutVertically(targetOffsetY = { it }) + fadeOut()
        ) {
            current?.let { data ->
                WfmToastContent(data = data)
            }
        }
    }
}
